hypothesis-python/tests/nocover/test_characters.py

```diff
@@ -32,3 +32,13 @@ def test_arbitrary_blacklist(data):
         )
     )
     assert c not in blacklist
+
+
+@given(st.characters(codec="ascii"))
+def test_ascii_codec(c):
+    assert c.encode(encoding="ascii").decode(encoding="ascii") == c
+
+
+@given(st.characters(codec="utf-8"))
+def test_utf8_codec(c):
+    assert c.encode().decode() == c
```

Comment:
We'll want to move these to `/cover/` tests so that we cover the new logic!  I'd also write them with a pytest.mark.parametrize over the different codec names, and then use the `st.data()` strategy to draw from strategies inside the test.